posts/how-to-build-a -travel-route-planning-application.md

```diff
@@ -0,0 +1,231 @@
+| [(:City {name: "Paris"}), (:City {name: "Heerlen"}), (:City {name: "Hannover"}), (:City {name: "Berlin"})] | 1055.63       |
++------------------------------------------------------------------------------------------------------------+---------------+
+```
+As you can see, the syntax is quite similar to breadth-first search syntax. Instead of a filter lambda, we need to provide a weight lambda and the total weight symbol. Given an edge and vertex pair, the weight lambda must return the cost of expanding to the given vertex using the given edge. The path returned will have the smallest possible sum of costs and it will be stored in the total weight symbol. A limitation of Dijkstra's algorithm is that the cost can't be negative.
+
+Let’s now assume you want to find the quickest route for your biking road trip where you don’t want to travel more than 200 km between stops. Just like you did for the breadth-first search query, you can add filters to the lambda function. In this case, you will need to specify that the length for every road `e.length` must be smaller than 200 km. 
+
+To get your result, you'll run this query:
+
+```
+MATCH p = (:City {name: "Paris"})
+      -[:Road * wShortest (e, v | e.length) total_weight (e, v | e.length <= 200)]->
+      (:City {name: "Berlin"})
+RETURN nodes(p) as cities, total_weight;
+```
+Your result should be:
+```
++-----------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------+--------------+
+| cities                                                                                                                                                                                                                | total_weight |
++-----------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------+--------------+
+| [(:City {name: "Paris"}), (:City {name: "Reims"}), (:City {name: "Charleroi"}), (:City {name: "Heerlen"}), (:City {name: "Hamm"}), (:City {name: "Hannover"}), (:City {name: "Magdeburg"}), (:City {name: "Berlin"})] | 1105.72      |
++-----------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------+--------------+
+```
+
+## Conclusion
+Congratulations! You just built a simple routing application by leveraging Memgraph and Cypher. Along the way, you learned how to use the breadth-first search algorithm and Dijkstra’s algorithm to navigate a complex network of inter-connected data using Cypher queries and filtered your results using the powerful and flexible lambda function.
```

Comment:
What do you think about asking a question at the end to provoke proactivity?

E.g. `Based on your travel planning experience, what other Cypher queries can you come up with that would be helpful for future travel planning?`